### PR TITLE
Format code to allow staticcheck linter to pass

### PR DESCRIPTION
**Pull request by stevemcquaid**
`staticcheck -fail -tests -checks="all,-ST1000,-ST1021,-ST1020" ./...`
 & 
`golangci-lint run --deadline=30m --disable-all --no-config --issues-exit-code=1 --enable=bodyclose --enable=deadcode --enable=dupl --enable=errcheck --enable=gocognit --enable=goconst  --enable=gocyclo --enable=gofmt --enable=goimports --enable=golint --enable=gomodguard --enable=gosec  --enable=govet --enable=ineffassign --enable=interfacer  --enable=megacheck --enable=misspell --enable=nakedret --enable=prealloc --enable=rowserrcheck --enable=staticcheck --enable=structcheck  --enable=stylecheck --enable=typecheck --enable=unconvert  --enable=unparam --enable=varcheck --enable=whitespace`

Both pass now. 

Also had to rename package name to be more go idiomatic: `ballistics`

### CODE REVIEW

File: Drag.go [hunk 1]
```diff
@@ -1,874 +1,789 @@
...
+
+var gICurve = calculateCurve(gITable)
+
+var gSTable = gITable // tables are the same
+
+var gSCurve = calculateCurve(gSTable)
+
+func calculateCurve(dataPoints []DataPoint) []CurvePoint {
+	var curve []CurvePoint
+	var numPoints = len(dataPoints)
```
Comment (stevemcquaid) [inline Drag.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	numPoints := len(dataPoints)
```

File: Drag.go [hunk 2]
```diff
@@ -1,874 +1,789 @@
...
+	var i int
+	var x1, x2, x3, y1, y2, y3, a, b, c float64
+
+	curve = make([]CurvePoint, numPoints)
+	var rate = (dataPoints[1].B - dataPoints[0].B) / (dataPoints[1].A - dataPoints[0].A)
```
Comment (stevemcquaid) [inline Drag.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	rate := (dataPoints[1].B - dataPoints[0].B) / (dataPoints[1].A - dataPoints[0].A)
```

File: Projectile.go [hunk 3]
```diff
@@ -1,89 +1,87 @@
...
+// CreateProjectileWithDimensions creates the description of a projectile with dimensions (diameter and length)
+//
+// Dimensions are only required if you want to take into account projectile spin drift.
+// TwistInfo must be also set in this case.
+func CreateProjectileWithDimensions(ballisticCoefficient BallisticCoefficient, bulletDiameter unit.Distance, bulletLength unit.Distance, weight unit.Weight) Projectile {
+	return Projectile{ballisticCoefficient: ballisticCoefficient,
+		hasDimensions:  true,
+		bulletDiameter: bulletDiameter,
+		bulletLength:   bulletLength,
+		weight:         weight}
```
Comment (stevemcquaid) [inline Projectile.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	return Projectile{
		ballisticCoefficient: ballisticCoefficient,
		hasDimensions:        true,
		bulletDiameter:       bulletDiameter,
		bulletLength:         bulletLength,
		weight:               weight,
	}
```

File: TrajectoryCalculator.go [hunk 4]
```diff
@@ -1,284 +1,281 @@
...
+
+	"github.com/gehtsoft-usa/go_ballisticcalc/bmath/unit"
+	"github.com/gehtsoft-usa/go_ballisticcalc/bmath/vector"
+)
+
+const cZeroFindingAccuracy float64 = 0.000005
+const cMinimumVelocity float64 = 50.0
+const cMaximumDrop float64 = -15000
+const cMaxIterations int = 10
+const cGravityConstant float64 = -32.17405
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
const (
	cZeroFindingAccuracy float64 = 0.000005
	cMinimumVelocity     float64 = 50.0
	cMaximumDrop         float64 = -15000
	cMaxIterations       int     = 10
	cGravityConstant     float64 = -32.17405
)
```

File: TrajectoryCalculator.go [hunk 5]
```diff
@@ -1,284 +1,281 @@
...
+	v.maximumCalculatorStepSize = x
+}
+
+func (v TrajectoryCalculator) getCalculationStep(step float64) float64 {
+	step = step / 2 // do it twice for increased accuracy of velocity calculation and 10 times per step
+
+	var maximumStep float64 = v.maximumCalculatorStepSize.In(unit.DistanceFoot)
+	if step > maximumStep {
+		var stepOrder = int(math.Floor(math.Log10(step)))
+		var maximumOrder = int(math.Floor(math.Log10(maximumStep)))
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
		stepOrder := int(math.Floor(math.Log10(step)))
		maximumOrder := int(math.Floor(math.Log10(maximumStep)))
```

File: TrajectoryCalculator.go [hunk 6]
```diff
@@ -1,284 +1,281 @@
...
+		maximumCalculatorStepSize: unit.MustCreateDistance(1, unit.DistanceFoot),
+	}
+}
+
+// SightAngle calculates the sight angle for a rifle with scope height specified and zeroed using the ammo specified at
+// the range specified and under the conditions (atmosphere) specified.
+//
+// The calculated value is to be used as sightAngle parameter of the ShotParameters structure
+func (v TrajectoryCalculator) SightAngle(ammunition Ammunition, weapon Weapon, atmosphere Atmosphere) unit.Angular {
+	var calculationStep = v.getCalculationStep(unit.MustCreateDistance(10, weapon.Zero().ZeroDistance().Units()).In(unit.DistanceFoot))
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	calculationStep := v.getCalculationStep(unit.MustCreateDistance(10, weapon.Zero().ZeroDistance().Units()).In(unit.DistanceFoot))
```

File: TrajectoryCalculator.go [hunk 7]
```diff
@@ -1,284 +1,281 @@
...
+	}
+	return unit.MustCreateAngular(barrelElevation, unit.AngularRadian)
+}
+
+// Trajectory calculates the trajectory with the parameters specified
+func (v TrajectoryCalculator) Trajectory(ammunition Ammunition, weapon Weapon, atmosphere Atmosphere, shotInfo ShotParameters, windInfo []WindInfo) []TrajectoryData {
+	var rangeTo float64 = shotInfo.MaximumDistance().In(unit.DistanceFoot)
+	var step float64 = shotInfo.Step().In(unit.DistanceFoot)
+
+	var calculationStep = v.getCalculationStep(step)
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	calculationStep := v.getCalculationStep(step)
```

File: TrajectoryCalculator.go [hunk 8]
```diff
@@ -1,284 +1,281 @@
...
+
+	var deltaRangeVector, rangeVector, velocityAdjusted, velocityVector, windVector, gravityVector vector.Vector
+	var muzzleVelocity, velocity, barrelAzimuth, barrelElevation float64
+	var drag float64
+	var time, deltaTime float64
+	var maximumRange, nextRangeDistance float64
+	var bulletWeight float64 = ammunition.Bullet().BulletWeight().In(unit.WeightGrain)
+
+	var stabilityCoefficient = 1.0
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	stabilityCoefficient := 1.0
```

File: TrajectoryCalculator.go [hunk 9]
```diff
@@ -1,284 +1,281 @@
...
+	var calculateDrift bool
+
+	if weapon.HasTwist() && ammunition.Bullet().HasDimensions() {
+		stabilityCoefficient = calculateStabilityCoefficient(ammunition, weapon, atmosphere)
+		calculateDrift = true
+	}
+
+	var rangesLength = int(math.Floor(rangeTo/step)) + 1
+	var ranges = make([]TrajectoryData, rangesLength)
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	rangesLength := int(math.Floor(rangeTo/step)) + 1
	ranges := make([]TrajectoryData, rangesLength)
```

File: TrajectoryCalculator.go [hunk 10]
```diff
@@ -1,284 +1,281 @@
...
+
+	barrelAzimuth = 0.0
+	barrelElevation = shotInfo.SightAngle().In(unit.AngularRadian)
+	barrelElevation = barrelElevation + shotInfo.ShotAngle().In(unit.AngularRadian)
+	var alt0 float64 = atmosphere.Altitude().In(unit.DistanceFoot)
+	// var densityFactor, mach float64 = atmosphere.getDensityFactorAndMachForAltitude(alt0) // ineffassign
+	var currentWind int
+	var nextWindRange = 1e7
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	nextWindRange := 1e7
```

File: TrajectoryCalculator.go [hunk 11]
```diff
@@ -1,284 +1,281 @@
...
+		}
+
+		if rangeVector.X >= nextRangeDistance {
+			var windage float64 = rangeVector.Z
+			if calculateDrift {
+				windage += (1.25 * (stabilityCoefficient + 1.2) * math.Pow(time, 1.83) * twistCoefficient) / 12.0
+			}
+
+			var dropAdjustment = getCorrection(rangeVector.X, rangeVector.Y)
+			var windageAdjustment = getCorrection(rangeVector.X, windage)
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
			dropAdjustment := getCorrection(rangeVector.X, rangeVector.Y)
			windageAdjustment := getCorrection(rangeVector.X, windage)
```

File: TrajectoryCalculator.go [hunk 12]
```diff
@@ -1,284 +1,281 @@
...
+	return ranges
+}
+
+func calculateStabilityCoefficient(ammunitionInfo Ammunition, rifleInfo Weapon, atmosphere Atmosphere) float64 {
+	var weight float64 = ammunitionInfo.Bullet().BulletWeight().In(unit.WeightGrain)
+	var diameter float64 = ammunitionInfo.Bullet().BulletDiameter().In(unit.DistanceInch)
+	var twist float64 = rifleInfo.Twist().Twist().In(unit.DistanceInch) / diameter
+	var length float64 = ammunitionInfo.Bullet().BulletLength().In(unit.DistanceInch) / diameter
+	var sd = 30 * weight / (math.Pow(twist, 2) * math.Pow(diameter, 3) * length * (1 + math.Pow(length, 2)))
+	var fv = math.Pow(ammunitionInfo.MuzzleVelocity().In(unit.VelocityFPS)/2800, 1.0/3.0)
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	sd := 30 * weight / (math.Pow(twist, 2) * math.Pow(diameter, 3) * length * (1 + math.Pow(length, 2)))
	fv := math.Pow(ammunitionInfo.MuzzleVelocity().In(unit.VelocityFPS)/2800, 1.0/3.0)
```

File: TrajectoryCalculator.go [hunk 13]
```diff
@@ -1,284 +1,281 @@
...
+
+	var ft float64 = atmosphere.Temperature().In(unit.TemperatureFahrenheit)
+	var pt float64 = atmosphere.Pressure().In(unit.PressureInHg)
+	var ftp = ((ft + 460) / (59 + 460)) * (29.92 / pt)
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	ftp := ((ft + 460) / (59 + 460)) * (29.92 / pt)
```

File: TrajectoryCalculator.go [hunk 14]
```diff
@@ -1,284 +1,281 @@
...
+}
+
+func windToVector(shot ShotParameters, wind WindInfo) vector.Vector {
+	var sightCosine = math.Cos(shot.SightAngle().In(unit.AngularRadian))
+	var sightSine = math.Sin(shot.SightAngle().In(unit.AngularRadian))
+	var cantCosine = math.Cos(shot.CantAngle().In(unit.AngularRadian))
+	var cantSine = math.Sin(shot.CantAngle().In(unit.AngularRadian))
+	var rangeVelocity = wind.velocity.In(unit.VelocityFPS) * math.Cos(wind.direction.In(unit.AngularRadian))
+	var crossComponent = wind.velocity.In(unit.VelocityFPS) * math.Sin(wind.direction.In(unit.AngularRadian))
+	var rangeFactor = -rangeVelocity * sightSine
```
Comment (stevemcquaid) [inline TrajectoryCalculator.go]: **[golangci-lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
```suggestion
	sightCosine := math.Cos(shot.SightAngle().In(unit.AngularRadian))
	sightSine := math.Sin(shot.SightAngle().In(unit.AngularRadian))
	cantCosine := math.Cos(shot.CantAngle().In(unit.AngularRadian))
	cantSine := math.Sin(shot.CantAngle().In(unit.AngularRadian))
	rangeVelocity := wind.velocity.In(unit.VelocityFPS) * math.Cos(wind.direction.In(unit.AngularRadian))
	crossComponent := wind.velocity.In(unit.VelocityFPS) * math.Sin(wind.direction.In(unit.AngularRadian))
	rangeFactor := -rangeVelocity * sightSine
```

**Comment by nikolaygekht** [comment]
It does not want to join, but I'll do it bymyself and set you as a co-author in the commit, ok?

**Comment by nikolaygekht** [comment]
Reproduced PR in commit #1  (f277483e2f2d82cf31996e8e9318728bfaa3c8f4)
@stevemcquaid co-authored in the commit

**Pull request closed without merging**